src/xcode/ENA/ENA/Source/Developer Menu/Features/DMServerEnvironmentViewController.swift

```diff
@@ -99,7 +99,8 @@ class DMServerEnvironmentViewController: UIViewController, UIPickerViewDelegate,
 
 			let selectedRow = self.picker.selectedRow(inComponent: 0)
 			self.store.selectedServerEnvironment = self.serverEnvironment.availableEnvironments()[selectedRow]
-			self.updateCurrentEnviromentLabel()
+			self.updateCurrentEnvironmentLabel()
+			self.store.appConfigMetadata = nil
```

Comment:
Should we clear the whole store to avoid problems like this in the future?
Also, not in scope of your fix, but shouldn't we also reset the event store?

Reply:
I'm not actually sure if we should reset everything, we should probably talk to testing about it if this is desirable for them to reset everything on every environment switch. Would definitely help to avoid these kinds of problems in the future